CriteoPublisherSdk/Sources/CR_BidManager.m

```diff
@@ -179,6 +180,47 @@ - (void)prefetchBidsForAdUnits:(CR_CacheAdUnitArray *)adUnits {
          }];
 }
 
+- (void)fetchLiveBidForAdUnit:(CR_CacheAdUnit *)adUnit
+           bidResponseHandler:(CR_BidResponseHandler)responseHandler {
+  [self fetchLiveBidForAdUnit:adUnit
+           bidResponseHandler:responseHandler
+                   timeBudget:CRITEO_DEFAULT_LIVE_BID_TIME_BUDGET_IN_SECONDS];
+}
+
+- (void)fetchLiveBidForAdUnit:(CR_CacheAdUnit *)adUnit
+           bidResponseHandler:(CR_BidResponseHandler)responseHandler
+                   timeBudget:(NSTimeInterval)timeBudget {
+  [self.threadManager dispatchAsyncOnGlobalQueueWithTimeout:timeBudget
+      operationHandler:^void(void (^completionHandler)(dispatchWithTimeoutHandler)) {
+        [self fetchBidsForAdUnits:@[ adUnit ]
+               cdbResponseHandler:^(CR_CdbResponse *cdbResponse) {
+                 completionHandler(^(BOOL handled) {
+                   [self.threadManager dispatchAsyncOnMainQueue:^{
+                     if (!handled) {
+                       NSAssert(cdbResponse.cdbBids.count <= 1,
+                                @"During a live request, only one bid will be fetched at a time.");
+                       if (cdbResponse.cdbBids.count == 1) {
+                         CR_CdbBid *bid = cdbResponse.cdbBids[0];
+                         responseHandler(bid);
+                       } else {
+                         responseHandler(nil);
+                       }
+                     } else {
+                       [self cacheBidsFromResponse:cdbResponse];
+                     }
+                   }];
+                 });
+               }];
+      }
+      timeoutHandler:^(BOOL handled) {
+        if (!handled) {
+          [self.threadManager dispatchAsyncOnMainQueue:^{
+            responseHandler([self getBid:adUnit]);
```

Comment:
Won't this getBid will try to fetch a new one if nothing is in the cache ?
I think that this is not expected.

Reply:
Nice catch, thanks, indeed, `getBid` isn't only getting a bid ;p I will add extra commits to rename & fix this.

Reply:
Fixed by ac16fc8 then a16227d